src/common/meta/src/wal/kafka/topic_manager.rs

```diff
@@ -54,8 +127,60 @@ impl TopicManager {
     }
 
     /// Selects a batch of topics from the topic pool through the topic selector.
-    pub fn select_batch(&self, num_topics: usize) -> Vec<Topic> {
-        // TODO(niebayes): calls `select` to select a collection of topics in a batching manner.
-        vec!["tmp_topic".to_string(); num_topics]
+    pub fn select_batch(&self, num_topics: usize) -> Vec<&Topic> {
+        (0..num_topics)
+            .map(|_| self.topic_selector.select(&self.topic_pool))
+            .collect()
+    }
+
+    async fn restore_created_topics(kv_backend: &KvBackendRef) -> Result<Vec<Topic>> {
+        kv_backend
+            .get(CREATED_TOPICS_KEY.as_bytes())
+            .await?
+            .map(|key_value| serde_json::from_slice(&key_value.value).context(DecodeJsonSnafu))
+            .unwrap_or_else(|| Ok(vec![]))
```

Comment:
It ignores errors while deserializing the value. I think we should return an error instead.

Reply:
Changed to the following: 
```rust
    async fn restore_created_topics(kv_backend: &KvBackendRef) -> Result<Vec<Topic>> {
        kv_backend
            .get(CREATED_TOPICS_KEY.as_bytes())
            .await?
            .map_or_else(
                || Ok(vec![]),
                |key_value| serde_json::from_slice(&key_value.value).context(DecodeJsonSnafu),
            )
    }
```